src/commands/inputs.rs

```diff
@@ -0,0 +1,139 @@
+use anyhow::{Context, Result};
+use clap::Parser;
+use indexmap::IndexMap;
+use serde_json::{Value, json};
+use std::path::{Path, PathBuf};
+use url::Url;
+use wdl::{
+    analysis::{path_to_uri, types::Type},
+    ast::{
+        AstToken, Document, SyntaxKind, Visitor,
+        v1::{
+            self,
+            Expr::{self, Literal},
+        },
+    },
+    cli::analyze,
+    doc,
+    grammar::SyntaxTree,
+};
+
+#[derive(Parser, Debug)]
+#[command(about = "Generate input JSON from a WDL document", version, about)]
+pub struct InputsArgs {
+    #[arg(required = true)]
+    #[clap(value_name = "input path")]
+    pub document: String,
+
+    #[arg(short, long)]
+    #[clap(value_name = "workflow or task name")]
+    pub name: Option<String>,
+
+    #[arg(short, long)]
+    #[clap(value_name = "output path")]
+    pub output: Option<PathBuf>,
+
+    #[arg(short, long)]
+    #[clap(value_name = "literal defaults")]
+    pub literal_defaults: bool,
+
+    #[arg(short, long)]
+    #[clap(value_name = "override expressions")]
+    pub override_expressions: bool,
+}
+
+pub async fn generate_inputs(args: InputsArgs) -> Result<()> {
+    let results: Vec<wdl::analysis::AnalysisResult> =
+        analyze(args.document.as_str(), vec![], false, false).await?;
+
+    let uri: Url = Url::parse(args.document.as_str()).unwrap_or_else(|_| {
+        path_to_uri(args.document.as_str()).expect("file should be a local path")
+    });
+
+    let result = results
+        .iter()
+        .find(|r| **r.document().uri() == uri)
+        .context("failed to find document in analysis results")?;
+
+    let document = result.document();
+
+    let diagnostics = document.diagnostics();
+    if !diagnostics.is_empty() {
```

Comment:
This should check for diagnostics of `Severity::Error` instead of just any diagnostic.